front/src/common/SubCategoryContext.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, useContext, useEffect, useMemo } from 'react';
+
+import { useTranslation } from 'react-i18next';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { SubCategory } from 'common/api/osrdEditoastApi';
+import { setFailure } from 'reducers/main';
+import { useAppDispatch } from 'store';
+
+type SubCategoryContextValue = {
+  subCategories: SubCategory[];
+};
```

Comment:
nit: maybe this type is not needed since it's so simple
```suggestion
type SubCategoryContextValue = SubCategory[];
```

Reply:
Done